source/common/stats/symbol_table_impl.h

```diff
@@ -130,22 +70,94 @@ class SymbolEncoding {
  */
 class SymbolTableImpl : public SymbolTable {
 public:
+  /**
+   * Intermediate representation for a stat-name. This helps store multiple names
+   * in a single packed allocation. First we encode each desired name, then sum
+   * their sizes for the single packed allocation. This is used to store
+   * MetricImpl's tags and tagExtractedName. Like StatName, we don't want to pay
+   * a vptr overhead per object, and the representation is shared between the
+   * SymbolTable implementations, so this is just a pre-declare.
+   */
+  class Encoding {
+  public:
+    /**
+     * Before destructing SymbolEncoding, you must call moveToStorage. This
+     * transfers ownership, and in particular, the responsibility to call
```

Comment:
Is this what the `ASSERT(vec_.empty());` in `SymbolTableImpl::Encoding::~Encoding()` checks for? If so, please add a comment at the callsite.